Tests/OpenTimelineIOTests/testExernalReference.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  testExternalReference.swift
+//
+
+import XCTest
+@testable import OpenTimelineIO
+
+import Foundation
+
+class tesExternalReference: XCTestCase {
+
+    func testURLs() {
+        for str in [
+            "/foo/bar.png",
+            "foo/bar.png",
+            "file:foo/bar.png",
+            "file:/foo/bar.png",
+            "file://foo/bar.png",
+            "file:///foo/bar.png",
+            "http:foo/bar.png",
+            "http:/foo/bar.png",
+            "http:www.foo.com/foo/bar.png",
+            "https:www.foo.com:1245/foo/bar.png",
+            "s3:foo/bar.png",
+            "s3:/foo/bar.png",
+            "s3://foo/bar.png",
+            "s3:///foo/bar.png",
+            "foo:/foo/bar.png?size=123&tag=&amp;&blah=bloog#1234"
```

Comment:
Can you add an empty string to this list also?

Reply:
Weirdly, NSURL disallows ``` URL(string: "") ``` It returns nil. Also weirdly, ``` URL(fileURLWithPath: "") ``` returns ``` file:///private/tmp/ ```

One possibility is to detect the empty string upon assignment and change it to a single dot, "."

Currently, nil is returned, in order to avoid a heuristic, which is why I made the accessor optional.

Does anyone have any thoughts about this conundrum?

Reply:
The extreme option would be to change the type of this from URL to String, and let the client code deal with it - but that seems not very friendly. Is the target_url field at the C++ level nullable? If yes, then it may be important to distinguish between null and empty string, but if not, then we could say that the Swift binding transforms a nil URL to the empty string in C++ and vice versa.

Reply:
I followed the same reasoning as you - target_url is a std::string, not a pointer to a string; this PR transforms a nil URL to an empty string, and returns nil for a C++ empty string. 

The further reasoning I followed is that the OTIO class ExternalReference returns a string, which can be empty, not a URL class of some sort. Since the Swift binding deviates by forming a Foundation URL, we've deviated from the underpinnings, and so we're not conforming to an OTIO standard, but a Foundation one. That was my second reason to return nil for empty, since URL itself can't be empty.

Reply:
Great. So when target_url is "" then the URL will be nil (and vice versa). That seems reasonable.